src/context/UserContext.jsx

```diff
@@ -0,0 +1,25 @@
+import React, { createContext, useEffect, useState } from "react";
+import { ApiClient } from "../api-client";
+
+export const UserContext = createContext();
+
+export const UserProvider = (props) => {
+  const [token, setToken] = useState(localStorage.getItem("awesomeTasksToken"));
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      const token_ = await ApiClient.fetchUser(token);
```

Comment:
Not clear what is `token` an `token_` and what is the difference. Maybe a bit more clear name would help here?